ts/Extensions/OverlappingDataLabels.ts

```diff
@@ -109,9 +109,13 @@ addEvent(Chart, 'render', function collectAndHide(): void {
                                 point.shapeArgs && point.shapeArgs.height
                             ); // #4118
 
-                            if (!options.allowOverlap) {
+                            if (
+                                !options.allowOverlap &&
+                                label.attr('visibility') !== 'hidden'
```

Comment:
If you decide to not push labels in the collector, then you need to do this for all labels (e.g. StackLabels, Radial, Annotations collectors). Additionally, someone must remember this in the future, to filter out hidden labels. I'm not sure what's the best. Filtering hidden labels in the `OvelappingLabels` module or preventing collecting hidden labels in the Collectors.

I will leave the decition up to @TorsteinHonsi

Reply:
PS: The stack labels already check visibility inside the collector. Which means if we do the check in the comparison itself, we can remove this:

https://github.com/highcharts/highcharts/blob/06dc2c70f9b04b32dd22857e6eabd2df0645f15b/ts/Extensions/OverlappingDataLabels.ts#L74